server/protocols/webadmin/webadmin-mailqueue/src/main/java/org/apache/james/webadmin/routes/MailQueueRoutes.java

```diff
@@ -74,8 +74,9 @@
     private final JsonTransformer jsonTransformer;
 
     @Inject
-    @VisibleForTesting MailQueueRoutes(MailQueueFactory<ManageableMailQueue> mailQueueFactory, JsonTransformer jsonTransformer) {
-        this.mailQueueFactory = mailQueueFactory;
+    @SuppressWarnings("unchecked")
+    @VisibleForTesting MailQueueRoutes(MailQueueFactory<?> mailQueueFactory, JsonTransformer jsonTransformer) {
+        this.mailQueueFactory = (MailQueueFactory<ManageableMailQueue>) mailQueueFactory;
```

Comment:
It looks like the consequences of not using the right generics in injection (and probably not using TypeLiteral). It's not important for now but should be removed in the future.

Reply:
I agree with you, the thing is that sometime we are waiting for a _MailQueueFactory<MailQueue>_ to be injected and other time for a _MailQueueFactory<ManageableMailQueue>_